projects/cashmere/src/lib/button-toggle/button-toggle-group.component.ts

```diff
@@ -0,0 +1,173 @@
+
+import { AfterContentInit, Component, ContentChildren, EventEmitter, HostBinding, Input, OnDestroy, Output, ViewEncapsulation } from '@angular/core';
+import { QueryList } from '@angular/core';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { ButtonToggleChangeEvent } from './button-toggle-change-event';
+import { parseBooleanAttribute } from '../util';
+import { ButtonToggleComponent } from './button-toggle.component';
+
+const supportedStyles = ['primary', 'primary-alt', 'destructive', 'neutral', 'secondary'];
+const supportedColors = ['blue', 'green', 'purple', 'red', 'orange', 'ruby-red', 'deep-red', 'red-orange', 'magenta', 'pink', 'light-pink', 'azure', 'teal', 'dark-green', 'brown', 'purple-gray', 'yellow', 'yellow-orange', 'tan'];
+const supportedSizes = ['sm', 'md', 'lg'];
+
+export function validateStyleInput(style: string) {
+    if (supportedStyles.indexOf(style) < 0 && supportedColors.indexOf(style) < 0) {
+        throw Error('Unsupported buttonStyle attribute value on ButtonToggleGroupComponent: ' + style);
+    }
+}
+
+export function validateSizeInput(size: string) {
+    if (supportedSizes.indexOf(size) < 0) {
+        throw Error('Unsupported size attribute value on ButtonToggleGroupComponent: ' + size);
+    }
+}
+
+/** `hc-button-toggle-group` components are on/off toggles with the appearance of an `hc-button`.
+ * These toggle groups may be configured to behave as single-select (like radio buttons), or multi-select (like checkboxes). */
+@Component({
+    selector: 'hc-button-toggle-group',
+    templateUrl: './button-toggle-group.component.html',
+    styleUrls: ['./button-toggle-group.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+export class ButtonToggleGroupComponent implements AfterContentInit, OnDestroy {
+    private _disabled = false;
+    private _style: string = 'secondary';
+    private _size: string = 'md';
+    private _valueRequired: boolean = false;
+    private _multiple: boolean = false;
+    private unsubscribe$ = new Subject<void>();
+
+    @HostBinding('class.hc-button-toggle-group') _hostClass = true;
+
+    @ContentChildren(ButtonToggleComponent)
+    _buttons: QueryList<ButtonToggleComponent>;
+
+    /** Event fired whenever a change is made to any button toggle in the group. */
+    @Output() selectionChangedEvent: EventEmitter<ButtonToggleChangeEvent> = new EventEmitter<ButtonToggleChangeEvent>();
+
+    /** Sets style of toggle. Choose from: `'primary' | 'primary-alt' | 'destructive' | 'neutral' | 'secondary'`.
+    * If needed, colors from the primary or secondary palette may be used as well (e.g. 'pink', 'red-orange', etc) */
+    @Input()
+    get buttonStyle(): string {
+        return this._style;
+    }
+    set buttonStyle(val: string) {
+        validateStyleInput(val);
+        if (supportedStyles.indexOf(val) < 0) {
```

Comment:
~This same line is called inside the function call on line 57. That function throws an error, but in here we just set it to a different string. What's the intended behavior here?~

Just ignore me 😆 I think I got it. This is just checking to see if it's a color and not a style, whereas the function checks if it's either one of those